adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,274 @@
+	}
+	if bidType == openrtb_ext.BidTypeVideo {
+		resetDigitalRequestData.Imps[0].MediaTypes.Video.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{*imp.Video.W, *imp.Video.H})
+	}
+
+	var extData = make(map[string]interface{})
+	err = json.Unmarshal(imp.Ext, &extData)
+	if err != nil {
+
+	} else {
+
+		resetDigitalRequestData.Imps[0].ZoneID.PlacementID = extData["bidder"].(map[string]interface{})["placement_id"].(string)
+		if resetDigitalRequestData.Imps[0].ZoneID.PlacementID == "test" {
+			resetDigitalRequestData.Imps[0].ForceBid = true
+		}
+
+	}
+	return resetDigitalRequestData
+
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, nil
+	}
+	if response.StatusCode != http.StatusOK {
+		return nil, nil
+	}
+
+	if err := json.Unmarshal(response.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(MaxBids)
+	//check no bids
+	jsonData := make(map[string]interface{})
+
+	json.Unmarshal([]byte(response.Body), &jsonData)
+	//Always one bid
+	bid := getBidFromResponse(jsonData)
+
+	bidType, err := getBidType(internalRequest.Imp[0])
+	if err != nil {
+		// handle error
+		return nil, []error{err}
+	}
+	bidResponse.Currency = getCurrency(internalRequest)
+	bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+		Bid:     bid,
+		BidType: bidType,
+	})
+
+	return bidResponse, nil
+}
+
+func getBidFromResponse(requestData map[string]interface{}) *openrtb2.Bid {
+	processData := requestData["bids"].([]interface{})[0].(map[string]interface{})
+
+	bid := &openrtb2.Bid{
+		ID:    processData["bid_id"].(string),
+		Price: getBidPrice(processData),
+		ImpID: processData["imp_id"].(string),
+		CrID:  processData["crid"].(string),
+	}
+	//if HTML is filled on jsonData then fill ADM with it
+	if value, ok := processData["html"].(string); ok {
+		bid.AdM = value
+	}
+	//if Width and Height are filled on jsonData then fill W and H with it
+	if value, ok := processData["w"].(string); ok {
+
+		i, _ := strconv.ParseInt(value, 10, 64)
+		if i > 0 {
+			bid.W = i
+		}
+	}
+	if value, ok := processData["h"].(string); ok {
+		i, _ := strconv.ParseInt(value, 10, 64)
+		if i > 0 {
+			bid.H = i
+		}
+	}
+	//if Bid Price is 0 then return nil
+	if bid.Price == 0 {
+		return nil
+	}
+
+	return bid
+}
+
+func getBidPrice(requestData map[string]interface{}) float64 {
+	if value, ok := requestData["cpm"].(float64); ok {
+		return value
+	}
+	return 0.0 // Default value if "cpm" doesn't exist or is not a float64
+}
+
+func getBidType(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
+	if imp.Banner != nil {
```

Comment:
Implemented as suggested

Reply:
> Implemented as suggested

could you point out or link where  `MType` changes are implemented?

Reply:
Sorry, it was addressed on the point that we support only single format bids, so we could assume the anti pattern. Anyway, it would be more advisable to change to the normal pattern?

Reply:
>Anyway, it would be more advisable to change to the normal pattern?

Prebid team recommends using [MType](https://github.com/prebid/openrtb/blob/main/openrtb2/bid.go#L334) field. But if it's not doable then current change suffices single format bid.  @bruno-siira  should mention in Bidder docs that adapter expects only single format bids in the incoming request

Reply:
When we're talking about the Bidder Docs what is this file exacly @onkarvhanumante

Reply:
@bruno-siira refer https://github.com/prebid/prebid-server/pull/3766/files#r1692702457